projects/packages/forms/src/blocks/contact-form/edit.js

```diff
@@ -72,332 +72,299 @@ const PRIORITIZED_INSERTER_BLOCKS = [ ...map( validFields, block => `jetpack/${
 const RESPONSES_PATH = `${ get( getJetpackData(), 'adminUrl', false ) }edit.php?post_type=feedback`;
 const CUSTOMIZING_FORMS_URL = 'https://jetpack.com/support/jetpack-blocks/contact-form/';
 
-export const JetpackContactFormEdit = forwardRef(
-	(
-		{
-			attributes,
-			setAttributes,
-			postAuthorEmail,
-			hasInnerBlocks,
-			replaceInnerBlocks,
-			selectBlock,
-			clientId,
-			instanceId,
-			className,
-			blockType,
-			variations,
-			defaultVariation,
-			canUserInstallPlugins,
-			style,
-		},
-		ref
-	) => {
-		const {
-			to,
-			subject,
-			customThankyou,
-			customThankyouHeading,
-			customThankyouMessage,
-			customThankyouRedirect,
-			jetpackCRM,
-			salesforceData,
-		} = attributes;
-		const [ isPatternsModalOpen, setIsPatternsModalOpen ] = useState( false );
-
-		const blockProps = useBlockProps();
-		const { isLoadingModules, isChangingStatus, isModuleActive, changeStatus } =
-			useModuleStatus( 'contact-form' );
+function JetpackContactFormEdit( { name, attributes, setAttributes, clientId, className } ) {
+	const {
+		to,
+		subject,
+		customThankyou,
+		customThankyouHeading,
+		customThankyouMessage,
+		customThankyouRedirect,
+		jetpackCRM,
+		salesforceData,
+	} = attributes;
+	const instanceId = useInstanceId( JetpackContactFormEdit );
+	const { replaceInnerBlocks, selectBlock } = useDispatch( 'core/block-editor' );
+	const {
+		blockType,
+		canUserInstallPlugins,
+		defaultVariation,
+		variations,
+		hasInnerBlocks,
+		postAuthorEmail,
+	} = useSelect(
+		select => {
+			const { getBlockType, getBlockVariations, getDefaultBlockVariation } =
+				select( 'core/blocks' );
+			const { getBlocks } = select( 'core/block-editor' );
+			const { getEditedPostAttribute } = select( 'core/editor' );
+			const { getUser, canUser } = select( 'core' );
+			const innerBlocks = getBlocks( clientId );
 
-		const formClassnames = clsx( className, 'jetpack-contact-form', {
-			'is-placeholder': ! hasInnerBlocks && registerBlockVariation,
-		} );
+			const authorId = getEditedPostAttribute( 'author' );
+			const authorEmail = authorId && getUser( authorId ) && getUser( authorId ).email;
+			const submitButton = innerBlocks.find( block => block.name === 'jetpack/button' );
+			if ( submitButton && ! submitButton.attributes.lock ) {
+				const lock = { move: false, remove: true };
+				submitButton.attributes.lock = lock;
+			}
 
-		const isSalesForceExtensionEnabled =
-			!! window?.Jetpack_Editor_Initial_State?.available_blocks[
-				'contact-form/salesforce-lead-form'
-			];
+			return {
+				blockType: getBlockType && getBlockType( name ),
```

Comment:
It seems there are some weird checks like this ( if selector is available ) and I believe it's just legacy code. The APIs checked are for years in WP and I believe it's safe to remove and simplify.

Another example is this [effect](https://github.com/Automattic/jetpack/pull/41274/files#diff-0048b4bd8ab00791237cc8a6b752332593a1e3d6e411b60e2d2fe983205eae72L143) that can be removed.